### ISSUE TITLE
Exception in thread "sbt-socket-server" java.lang.UnsatisfiedLinkError on arm64 platform

### ISSUE THREAD

**Issue by eeedean** [scala-js/scala-js#4622]
I'm trying a minimalistic example for Scala.js for getting started on my M1 Mac, which runs an ARM64 Chip. 

Hardware: MacBook Pro M1 Max (2021)
Operating System: macOS Monterey 12.1

JDK: Bellsofts Liberica JDK 11.0.13 (LTS)
sbt version: 1.3.7 (project) and 1.5.5 (sbt script)
Scala version: 2.13.1
scala-js sbt plugin version: 1.8.0

I was following the [basic tutorial of the Homepage](http://www.scala-js.org/doc/tutorial/basic/). However the first sbt call fails like that:
```
 ~/workspace/scala/scalajs-test ❯ sbt -v
[sbt_options] declare -a sbt_options='()'
[process_args] java_version = '11'
[copyRt] java9_rt = '/Users/edean/.sbt/1.0/java9-rt-ext-bellsoft_11_0_13/rt.jar'
# Executing command line:
java
-Dfile.encoding=UTF-8
-Xms1024m
-Xmx1024m
-Xss4M
-XX:ReservedCodeCacheSize=128m
-Dscala.ext.dirs=/Users/edean/.sbt/1.0/java9-rt-ext-bellsoft_11_0_13
-jar
/Users/edean/.sdkman/candidates/sbt/1.5.5/bin/sbt-launch.jar

[info] Loading settings for project global-plugins from credentials.sbt ...
[info] Loading global plugins from /Users/edean/.sbt/1.0/plugins
[info] Loading settings for project scalajs-test-build from plugins.sbt ...
[info] Loading project definition from /Users/edean/workspace/scala/scalajs-test/project
[info] Loading settings for project scalajs-test from build.sbt ...
[info] Set current project to Scala.js Tutorial (in build file:/Users/edean/workspace/scala/scalajs-test/)
Exception in thread "sbt-socket-server" java.lang.UnsatisfiedLinkError: /private/var/folders/r9/hwm307t10rq9n_fkmnrpwbth0000gn/T/jna-96354899/jna4459817499867818637.tmp: dlopen(/private/var/folders/r9/hwm307t10rq9n_fkmnrpwbth0000gn/T/jna-96354899/jna4459817499867818637.tmp, 0x0001): tried: '/private/var/folders/r9/hwm307t10rq9n_fkmnrpwbth0000gn/T/jna-96354899/jna4459817499867818637.tmp' (fat file, but missing compatible architecture (have 'i386,x86_64', need 'arm64e')), '/usr/lib/jna4459817499867818637.tmp' (no such file)
	at java.base/java.lang.ClassLoader$NativeLibrary.load0(Native Method)
	at java.base/java.lang.ClassLoader$NativeLibrary.load(ClassLoader.java:2442)
	at java.base/java.lang.ClassLoader$NativeLibrary.loadLibrary(ClassLoader.java:2498)
	at java.base/java.lang.ClassLoader.loadLibrary0(ClassLoader.java:2694)
	at java.base/java.lang.ClassLoader.loadLibrary(ClassLoader.java:2627)
	at java.base/java.lang.Runtime.load0(Runtime.java:768)
	at java.base/java.lang.System.load(System.java:1837)
	at com.sun.jna.Native.loadNativeDispatchLibraryFromClasspath(Native.java:947)
	at com.sun.jna.Native.loadNativeDispatchLibrary(Native.java:922)
	at com.sun.jna.Native.<clinit>(Native.java:190)
	at com.sun.jna.Pointer.<clinit>(Pointer.java:54)
	at com.sun.jna.Structure.<clinit>(Structure.java:2130)
	at sbt.internal.server.Server$$anon$1$$anon$2.$anonfun$run$1(Server.scala:65)
	at scala.util.Try$.apply(Try.scala:213)
	at sbt.internal.server.Server$$anon$1$$anon$2.run(Server.scala:60)
```

My Directory looks like that: 
```
.
├── build.sbt
├── project
│   ├── build.properties
│   └── plugins.sbt
└── src
    └── main
        └── scala
            └── tutorial
                └── webapp
                    └── TutorialApp.scala
```

My Files look like that: 
**build.sbt**
```scala
enablePlugins(ScalaJSPlugin)

name := "Scala.js Tutorial"
scalaVersion := "2.13.1" // or any other Scala version >= 2.11.12

// This is an application with a main method
scalaJSUseMainModuleInitializer := true
```

**project/build.properties**
```scala
sbt.version=1.3.7
```

**project/plugins.sbt**
```scala
addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.8.0")
```

**src/main/scala/tutorial/webapp/TutorialApp.scala**
```scala
package tutorial.webapp

object TutorialApp {
  def main(args: Array[String]): Unit = {
    println("Hello world!")
  }
}
```

---

Obviously it fails due to a missing arm64 version for a native binary. However, I did not manage to figure out, which is failing at this point, due to missing experience with the toolchain. Nevertheless, I'm looking forward to provide additional information and try, whatever comes to mind. 

Thanks for all your efforts, you rock!

**Comment by keynmol** [comment]
For what it's worth, on an M1 with same versions (admitedly old-ish Scala and old-ish SBT) I also get that exception **only** with a `-v` flag to SBT.

And 10s after that exception it outputs `[warn] sbt server could not start in 10s` and does drop into an SBT shell.

Running `sbt run` works. 

Updating SBT version to 1.6.1 in `build.properties` makes the exception disappear when running `sbt -v`

**Comment by keynmol** [comment]
```
❯ find . | ack '\.(sbt|scala|properties)$' | xargs -n1 -I{} sh -c 'echo "\n\n# {}" && cat {}'


# ./project/build.properties
sbt.version=1.6.1


# ./project/plugins.sbt
addSbtPlugin("org.scala-js" % "sbt-scalajs" % "1.8.0")


# ./build.sbt
enablePlugins(ScalaJSPlugin)

name := "Scala.js Tutorial"
scalaVersion := "2.13.1" // or any other Scala version >= 2.11.12

// This is an application with a main method
scalaJSUseMainModuleInitializer := true


# ./src/main/scala/TutorialApp.scala
package tutorial.webapp

object TutorialApp {
  def main(args: Array[String]): Unit = {
    println("Hello world!")
  }
}
```

```
❯ java -version
openjdk version "17" 2021-09-14
OpenJDK Runtime Environment Temurin-17+35 (build 17+35)
OpenJDK 64-Bit Server VM Temurin-17+35 (build 17+35, mixed mode)
```

```
❯ sbt -v run
[sbt_options] declare -a sbt_options='()'
[process_args] java_version = '17'
[copyRt] java9_rt = '/Users/velvetbaldmime/.sbt/1.0/java9-rt-ext-eclipse_adoptium_17/rt.jar'
# Executing command line:
java
-Dfile.encoding=UTF-8
-Xms1024m
-Xmx1024m
-Xss4M
-XX:ReservedCodeCacheSize=128m
-Dsbt.script=/Users/velvetbaldmime/.tools/sbt
-Dscala.ext.dirs=/Users/velvetbaldmime/.sbt/1.0/java9-rt-ext-eclipse_adoptium_17
-jar
/Users/velvetbaldmime/.cache/sbt/boot/sbt-launch/1.6.0-RC2/sbt-launch-1.6.0-RC2.jar
run

[info] welcome to sbt 1.6.1 (Eclipse Adoptium Java 17)
[info] loading settings for project tmp-maazk16e-build from plugins.sbt ...
[info] loading project definition from /private/var/folders/my/drt5584s2w59ncgxgxpq7v040000gn/T/tmp.Maazk16e/project
[info] loading settings for project tmp-maazk16e from build.sbt ...
[info] set current project to Scala.js Tutorial (in build file:/private/var/folders/my/drt5584s2w59ncgxgxpq7v040000gn/T/tmp.Maazk16e/)
[info] compiling 1 Scala source to /private/var/folders/my/drt5584s2w59ncgxgxpq7v040000gn/T/tmp.Maazk16e/target/scala-2.13/classes ...
[info] Non-compiled module 'compiler-bridge_2.13' for Scala 2.13.1. Compiling...
7 warnings found
[info]   Compilation completed in 3.133s.
[info] Fast optimizing /private/var/folders/my/drt5584s2w59ncgxgxpq7v040000gn/T/tmp.Maazk16e/target/scala-2.13/scala-js-tutorial-fastopt
[info] Running tutorial.webapp.TutorialApp.
Hello world!
[success] Total time: 5 s, completed 12 Jan 2022, 22:35:02
```

**Comment by eeedean** [comment]
I did not expect to have an answer this fast! Glad, there's at least one expert running arm silicon, too. 

I can verify, that it does indeed work with project sbt version 1.6.1, and `sbt run` indeed worked just fine in the first place! Sorry, for not testing that initially. Rookie move, I guess. 

Also I just observed, running under Rosetta2 works out just fine, too (with 1.3.7 project sbt version, even running `sbt` alone).

**So best solution seems to be project sbt version 1.6.1 to me. **

I'll better get on my first steps again, then. 
Thanks so much!
